packages/slice-machine/src/hooks/useServerState.tsx

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from "react";
+
+import { ServerState } from "@lib/models/server/ServerState";
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+
+const useServerState = (serverState: ServerState | undefined) => {
+  const { getEnvironment } = useSliceMachineActions();
+
+  useEffect(() => getEnvironment(serverState), [serverState]);
```

Comment:
Not sure to understand why we need to update the Environment when the State changes.
I'm talking from the last project where I did the same and It ended up badly with useEffect depending on each other.
I'm not asking you to change, just let's be careful not to have more useEffect depending on each other